dask/dataframe/partitionquantiles.py

```diff
@@ -382,7 +388,7 @@ def process_val_weights(vals_and_weights, npartitions, dtype_info):
         rv = pd.DatetimeIndex(rv).tz_localize(dtype.tz)
     elif "datetime64" in str(dtype):
         rv = pd.DatetimeIndex(rv, dtype=dtype)
-    elif rv.dtype != dtype:
+    elif str(rv.dtype) != str(dtype):
```

Comment:
Just curious, why this change?

Reply:
If I understand correctly, older versions of numpy don't have support for pandas extension dtypes, and so doing this comparison when the left value is a numpy dtype will raise a `TypeError` (in our dataframe mindeps build):

```
>>> np.dtype("float64") == pd.Int64Dtype()
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
TypeError: data type not understood
```

Casting to strings seemed like a simple workaround, though I'm not sure if this is an indication that we should bump our numpy min version for `dask[dataframe]` if we continue to support extension dtypes more broadly.

Reply:
I see, thanks for the explanation. Do you know what version of `numpy` we would need to bump the minimum to?

Reply:
Looks like we would need to bump to `numpy=1.21` for the equality check above not to error, which in turn would force us to bump the Pandas min version so that NumPy doesn't emit warnings; knowing this, maybe it's better not to bump things for now.

However, looking through the code it looks like `rv.dtype` _should_ always be a `numpy.dtype`, so if we switch the order of the inequality check:

```suggestion
    elif dtype != rv.dtype:
```

We end up using `dtype.__ne__` instead, which in the case of pandas extension dtypes does have support for comparing against numpy dtypes and should resolve the failures.

Reply:
> Looks like we would need to bump to numpy=1.21 for the equality check above not to error, which in turn would force us to bump the Pandas min version so that NumPy doesn't emit warnings; knowing this, maybe it's better not to bump things for now.

Totally agree we can hold off on bumping things here -- definitely okay having that be out of scope for this PR. Worth considering as a follow-up though. It's been a while since we've updated minimum versions. 

> We end up using dtype.__ne__ instead, which in the case of pandas extension dtypes does have support for comparing against numpy dtypes and should resolve the failures.

Nice detective work. If it works here, and doesn't break any existing tests, then that sounds good to me